api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java

```diff
@@ -4098,4 +4098,162 @@ public CohortDefinition getMQC13P3NUM_I() {
 
     return sqlCohortDefinition;
   }
+
+  /**
+   * <b>MQ13</b>: Melhoria de Qualidade Category 13 Part 4, B2 <br>
+   *
+   * <ul>
+   *   <li>B2 - Select all patients from Ficha Clinica (encounter type 6) with FIRST concept “Carga
+   *       Viral” (Concept id 856) with value_numeric > 1000 and Encounter_datetime during the
+   *       Inclusion period (startDateInclusion and endDateInclusion). Note: if there is more than
+   *       one record with value_numeric > 1000 than consider the first occurrence during the
+   *       inclusion period.
+   *   <li>
+   * </ul>
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getMQ13P4B() {
+    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
+    sqlCohortDefinition.addParameter(new Parameter("startDate", "Start date", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("endDate", "End date", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("location", "Location", Date.class));
+
+    sqlCohortDefinition.setName("Category 13 - Part 4 Denominator - B2");
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map.put("856", hivMetadata.getHivViralLoadConcept().getConceptId());
+
+    String query =
+        "SELECT  "
+            + "     patient_id  "
+            + " FROM  "
+            + "     (SELECT   "
+            + "         p.patient_id, MIN(e.encounter_datetime)  "
+            + "     FROM  "
+            + "         patient p  "
+            + "     INNER JOIN encounter e ON e.patient_id = p.patient_id  "
+            + "     INNER JOIN obs o ON o.encounter_id = e.encounter_id  "
+            + "     WHERE  "
+            + "         p.voided = 0 AND e.voided = 0  "
+            + "             AND o.voided = 0  "
+            + "             AND e.encounter_type = ${6}  "
+            + "             AND e.location_id = :location  "
+            + "             AND e.encounter_datetime BETWEEN :startDate AND :endDate  "
+            + "             AND o.concept_id = ${856}  "
+            + "             AND o.value_numeric > 1000  "
```

Comment:
@amosh101 this check should not be here, this will make the querie to look only that concept where the value_numeric is >1000, you need to get first the minimal encounter that you have the concept 856 then you will see if the value is > 1000

Reply:
Done.